examples/src/examples/graphics/contact-hardening-shadows.tsx

```diff
@@ -0,0 +1,367 @@
+                });
+                camera.setLocalPosition(0, 5, 11);
+
+                camera.camera.requestSceneColorMap(true);
+                camera.addComponent("script");
+                camera.script.create("orbitCamera", {
+                    attributes: {
+                        inertiaFactor: 0.2,
+                        focusEntity: occluder,
+                        distanceMax: 500,
+                        frameOnStart: false
+                    }
+                });
+                camera.script.create("orbitCameraInputMouse");
+                camera.script.create("orbitCameraInputTouch");
+                app.root.addChild(camera);
+
+                data.on('*:set', (path: string, value: any) => {
+                    if (path === 'script.area.intensity') {
+                        areaLight.light.intensity = value;
+                        brightMaterial.emissiveIntensity = value;
+                        brightMaterial.update();
+                    } else if (path === 'script.area.size') {
+                        areaLight.light.lightSize = value;
+                    } else if (path === 'script.area.shadowType') {
+                        areaLight.light.shadowType = parseInt(value);
+                    } else if (path === 'script.directional.intensity') {
+                        directionalLight.light.intensity = value;
+                    } else if (path === 'script.directional.size') {
+                        directionalLight.light.lightSize = value;
+                    } else if (path === 'script.directional.shadowType') {
+                        directionalLight.light.shadowType = parseInt(value);
+                    } else if (path === 'script.point.intensity') {
+                        lightOmni.light.intensity = value;
+                    } else if (path === 'script.point.size') {
+                        lightOmni.light.lightSize = value;
+                    } else if (path === 'script.point.shadowType') {
+                        lightOmni.light.shadowType = parseInt(value);
+                    } else if (path === 'light.distance') {
+                        areaLight.setLocalPosition(0, 4, -value);
+                    }
+                });
+
+                let resizeControlPanel = true;
+                let time = 0;
+                let timeDiff = 0;
+                let index = 0;
+                app.on("update", function (dt) {
+
+                    if (time === 0) {
+                        // @ts-ignore engine-tsd
+                        camera.script.orbitCamera.distance = 25;
+                    }
+                    time += dt;
+                    timeDiff += dt;
+
+                    if (data.get('script.cycle')) {
+                        if ((timeDiff / 5) > 1) {
+                            index = (index + 1) % 3;
+                            timeDiff = 0;
+                        }
+    
+                        areaLight.enabled = false;
```

Comment:
we do not want disable lights every frame, just to enable one later, as that is pretty expensive under the hood. Ideally this is something like:
```
areaLight.enabled = index === 0;
directionalLight.enabled = index === 1;
lightOmni.enabled = index === 2;
```

I know it's just an example, but we should show how to do little things properly as well.